docs/errors.md

```diff
@@ -1,41 +1,267 @@
-# Errors
+# MSAL.js errors
 
----
+## Common errors
```

Comment:
nit: I would suggest rather than grouping by library, which is opaque to customers, we group these by the error classes themselves e.g. `BrowserAuthError`, `ClientConfigurationError`, etc.

Reply:
+1. The categorization is always preferred based on type and not the layer.

Reply:
done